airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -65,6 +65,10 @@ class PodLaunchFailedException(AirflowException):
     """When pod launching fails in KubernetesPodOperator."""
```

Comment:
[merge blocker]  If we like how I implemented this, I'll go ahead and apply the same changes to the other waiters.  Please do not merge this.